File: types/patchstatus.go
package types

type PatchInstanceInfo struct {
	Identifier        string   `json:"identifier"`
	IsWriter          bool     `json:"is_writer"`
	MaintenanceWindow string   `json:"maintenance_window"`
	PendingActions    []string `json:"pending_actions"`
}

type PatchStatus struct {
	Active             bool                `json:"active"`
	AutoPatchEnabled   bool                `json:"auto_patch_enabled"`
	TotalInstances     int                 `json:"total_instances"`
	PatchedCount       int                 `json:"patched_count"`
	CurrentlyPatching  string              `json:"currently_patching"`
	TempInstanceName   string              `json:"temp_instance_name"`
	PendingInstances   []PatchInstanceInfo `json:"pending_instances"`
	CompletedInstances []string            `json:"completed_instances"`
}

File: ui/src/types/PatchStatus.ts
interface PatchInstanceInfo {
    identifier: string;
    is_writer: boolean;
    maintenance_window: string;
    pending_actions: string[];
}

interface PatchStatus {
    active: boolean;
    auto_patch_enabled: boolean;
    total_instances: number;
    patched_count: number;
    currently_patching: string;
    temp_instance_name: string;
    pending_instances: PatchInstanceInfo[];
    completed_instances: string[];
}
